.github/workflows/ci.yml

```diff
@@ -14,16 +14,18 @@ jobs:
     outputs:
       workspaces: ${{ steps.find-changed-workspaces.outputs.workspaces }}
     steps:
-      - name: Checkout base branch for diff purposes
-        uses: actions/checkout@1d96c772d19495a3b5c517cd2bc0cb401ea0529f # v4
-        with:
-          ref: ${{ github.event.pull_request.base.ref }}
-
       - name: Calculate number of commits in PR
         id: calculate-commits
         env:
           COMMITS: ${{ github.event.pull_request.commits }}
         run: echo "NUMBER_OF_COMMITS=$(($COMMITS + 1))" >> $GITHUB_ENV
+
+      - name: Checkout base branch for diff purposes
+        uses: actions/checkout@1d96c772d19495a3b5c517cd2bc0cb401ea0529f # v4
+        with:
+          ref: ${{ github.event.pull_request.base.ref }}
+          fetch-depth: ${{ env.NUMBER_OF_COMMITS }}
```

Comment:
If we're looking for a stop-gap solution to iterate on, I'd just hard-code a number > 1. I don't think that the number of commits on a PR branch correlates to how outdated a branch root is. My 50 below is based on not a whole lot, but looking at https://github.com/backstage/community-plugins/graphs/commit-activity, that covers a week or more for most weeks.

```suggestion
          fetch-depth: 50
```

Reply:
Ok, updated to 50.